### PR TITLE
NEW: queque next episodes when playing from library

### PR DESCRIPTION
**Pull request by bcedu**
# Binge mode

When a file is played from the library, all next episodes are added to the queue. By this, users cab star to watch a show without the need of select the next episode every time an episode finish.

imgl1

imgl2

## Other changes

- When a file is played from the library, the playlist is cleared before adding all the next episodes of the selected file.

### CODE REVIEW

**Comment by bcedu** [comment]
@danrabbit 
> Just first impressions, it looks like this branch isn't just about automatically playing the next file in the queue. This seems to try to automatically find more video files in the parent directory to add to the queue. So I think the tooltip needs to be a little more explicit about that

I've updated the description and added a label to the button to make it clear.

> Especially since there doesn't seem to be any kind of trying to parse file names or metadata to make sure the video is actually related, it just adds anything in the parent folder. This could lead to embarrassing situations

Actually the files are quequed only if they are videos:
`if (newfile.query_info("*", 0).get_content_type().split("/")[0] == "video")`

> Using `media-playback-stop-symbolic` here is probably not appropriate since this is an action icon and it's a bit confusing. I'm not sure it makes sense for a user to click on a "stop" icon in order to add new things to the queue. This might require new bespoke icons.

I've changed to a standard toggle button with a description. I think it's less confusing for the user.

> I think if we're approaching this like a setting that it should be persistent between sessions, but I wonder if it would be better as a one time action button instead of a toggle.

I've changed the behaviour. When this option is activation a file is added to the queque. Also when a file ends its playback another file is added to the queque. By doing this the user always see the next file that will be played instead of playing it directlly.

File: src/Widgets/Player/PlaylistPopover.vala [hunk 1]
```diff
@@ -40,6 +41,9 @@ public class Audience.Widgets.PlaylistPopover : Gtk.Popover {
         rep.set_image (new Gtk.Image.from_icon_name ("media-playlist-no-repeat-symbolic", Gtk.IconSize.BUTTON));
         rep.set_tooltip_text (_("Enable Repeat"));
 
+        Gtk.Label autoqueque_next_label = new Gtk.Label (_("Auto queque aviable videos:"));
```
Comment (cassidyjames) [inline src/Widgets/Player/PlaylistPopover.vala]:
```suggestion
        Gtk.Label autoqueque_next_label = new Gtk.Label (_("Auto queue available videos:"));
```

**Comment by bcedu** [comment]
@cassidyjames @danrabbit Any news? If you are not interessted in adding this feature i will just merge it in my fork, no problem :+1:

**Comment by cassidyjames** [comment]
@bcedu I like the idea of a queue, I'm just a little hesitant with this implementation; I like the ability to have a queue and to be able to see upcoming videos in the queue, but I don't think adding them automatically from the filesystem is desired, especially since the order of those videos could feel nondeterministic if it's based on filename (`The Phantom Menace.mp4` should come before `Attack of the Clones.mp4`, but an automatic queuing system wouldn't know that). 

I think what Dan was implying about an embarrassing situation is if you opened a file in your Downloads folder to watch, then it automatically started adding _other_ videos from your Downloads folder into the queue, that might not be something that you'd want to happen. It could be avoided altogether by making queuing explicitly opt-in and controlled by the user from the UI or by explicitly opening multiple files (or an entire folder) with the Videos app.

**Comment by bcedu** [comment]
@cassidyjames What if i change to "Play next video from library". If the user explore the library and plays a video, it would have the option to automaticly play/quque the next video of the library. By this, I would not queque videos directly from the filesystem.

PD: I'm just trying to imitate the "autoplay" behaviour of Netflix. Just lay in the sofa and let the app play the next video of my favorite tv show.

**Comment by cassidyjames** [comment]
@bcedu ah, binge mode! I think that makes sense with TV series from the library, yes. I am less sure it works for arbitrary videos or movies, though. Do you know if we currently make that distinction in the code anywhere? I think we display them differently based on whether they're in the top-level folder (movies) or in subfolders (one per show?).

**Comment by bcedu** [comment]
> @bcedu ah, binge mode! I think that makes sense with TV series from the library, yes. I am less sure it works for arbitrary videos or movies, though. Do you know if we currently make that distinction in the code anywhere? I think we display them differently based on whether they're in the top-level folder (movies) or in subfolders (one per show?).

@cassidyjames 
I'm don't know well the structure followed by the library, but for what i've seen, there is no distinction. What i've finally made is that when a file is played from the library, **all the next files** aviable in the current library page **are added to the playlist**. By this, I don't add arbritary files to the playlist, i just add files that are already in the library. I also add the files in the same order that they are showed to the user in the library.

If you want, I could add a config button to activat and deactivate this behaviour.

**Comment by cassidyjames** [comment]
@bcedu did you mean to close this? I'd still be interested in a TV-inspired binge mode for series.

**Comment by bcedu** [comment]
@cassidyjames i closed it because I had no response and i thought you were not interested. But if you are still interested we can reopen it 😁

**Comment by cassidyjames** [comment]
@bcedu definitely still interested in the feature! We've just been busy with other things as well.

**Comment by bcedu** [comment]
> @bcedu definitely still interested in the feature! We've just been busy with other things as well.

@cassidyjames 
I've made a new PR with this changes: https://github.com/elementary/videos/pull/151

**Pull request closed without merging**